lib/ae_mdw/activities.ex

```diff
@@ -165,15 +166,37 @@ defmodule AeMdw.Activities do
         end
         |> Collection.paginate(pagination)
 
-      events =
-        Enum.map(activities_locators_data, fn {{height, txi, _local_idx}, data} ->
-          render(state, account_pk, height, txi, data)
-        end)
+      events = render_activities(state, account_pk, activities_locators_data)
 
       {:ok, serialize_cursor(prev_cursor), events, serialize_cursor(next_cursor)}
     end
   end
 
+  defp render_activities(state, account_pk, activities_locators_data) do
+    {activities_locators_data, _acc} =
+      Enum.map_reduce(activities_locators_data, {nil, nil}, fn
+        {{height, _txi, _local_idx}, _data} = locator, {height, enc_kb_hash} = kb_info ->
+          {{enc_kb_hash, locator}, kb_info}
+
+        {{height, _txi, _local_idx}, _data} = locator, _kb_info ->
+          Model.block(hash: kb_hash) = State.fetch!(state, Model.Block, {height, -1})
+          enc_kb_hash = Enc.encode(:key_block_hash, kb_hash)
+
+          {{enc_kb_hash, locator}, {height, enc_kb_hash}}
+      end)
+
+    Enum.map(activities_locators_data, fn {key_block_hash, {{height, txi, _local_idx}, data}} ->
+      {type, payload} = render_payload(state, account_pk, height, txi, data)
+
+      %{
+        height: height,
+        key_block_hash: key_block_hash,
```

Comment:
Depending on the requirement for key block information, instead of `key_block_hash`, or additionally to it, could be the `block_hash` with the micro block like `/txs`, to indicate the exact its location.

Reply:
This is a good idea, I'm going to implement it this way

Reply:
Done